src/transaction/flashback_cl.c

```diff
@@ -0,0 +1,196 @@
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * flashback_cl.c -
+ */
+
+#ident "$Id$"
+
+#include "config.h"
+
+#include <stdio.h>
+#if defined(WINDOWS)
+#include <windows.h>
+#else /* WINDOWS */
+#include <sys/ioctl.h>
+#include <unistd.h>
+#endif /* LINUX */
+
+#include "flashback_cl.h"
+#include "object_representation.h"
+
+#if defined(WINDOWS)
+static int
+flashback_util_get_winsize ()
+{
+  CONSOLE_SCREEN_BUFFER_INFO csbi;
+
+  GetConsoleScreenBufferInfo (GetStdHandle (STD_OUTPUT_HANDLE), &csbi);
+
+  return csbi.srWindow.bottom - csbi.srWindow.Top + 1;
+}
+#else
+static int
+flashback_util_get_winsize ()
+{
+  struct winsize w;
+
+  ioctl (STDOUT_FILENO, TIOCGWINSZ, &w);
+
+  return w.ws_row;
+}
+#endif
+
+/*
+ * flashback_find_class_index () - find the index of oidlist which contains classoid
+ *
+ * return              : class index
+ * classname_list (in) : classnames
+ * oidlist (in)        : OID of classnames
+ * classoid (in)       : classoid to find
+ */
+
+static int
+flashback_find_class_index (OID * oidlist, int list_size, OID classoid)
+{
+  int class_index = 0;
+
+  for (class_index = 0; class_index < list_size; class_index++)
+    {
+      if (OID_EQ (&oidlist[class_index], &classoid))
+	{
+	  return class_index;
+	}
+    }
+
+  return -1;
+}
+
+/*
+ * flashback_unpack_and_print_summary () 
+ *
+ * return                   : error code
+ * summary_buffer (in/out)  : input buffer which contains summary info, and return advanced buffer pointer
+ * summary (out)            : brief summary information
+ * num_summary (in)         : number of summary to unpack
+ * classname_list (in)      : classnames
+ * oidlist (in)             : OID of classnames
+ */
+
+int
+flashback_unpack_and_print_summary (char **summary_buffer, FLASHBACK_SUMMARY_INFO_MAP * summary, int num_summary,
+				    dynamic_array * classname_list, OID * oidlist)
+{
+  TRANID trid;
+  char *user = NULL;
+  time_t start_time, end_time;
+  int num_insert, num_update, num_delete;
+  LOG_LSA start_lsa, end_lsa;
+  int num_table;
+  OID classoid;
+  char *tmp_ptr = *summary_buffer;
+
+  bool stop_print = false;
+  const int max_window_size = flashback_util_get_winsize ();
+  int line_cnt = 0;
+
+  printf
+    ("  transaction id  |    user   |   start time    |   end time   |   num_insert   |   num_update   |   num_delete   |   tables   ");
+
+  line_cnt++;
```

Comment:
Why the `line_cnt` is not 1?
Are there any reasons that `line_cnt` should be  initialized 0?

Reply:
Since no lines were initially printed on the screen, line_cnt is initialized to 0. 
It is more intuitive to increment line_cnt after every line is printed, even if it initially outputs one line unconditionally.

Reply:
Oh, I see. Thank you for the explaination.